src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java

```diff
@@ -51,37 +45,27 @@ public ParallelFlowExecutorServiceImpl(ExecutorService threadPoolExecutor,
      */
     @Override
     public void execute() {
-        while (FLAG) {
-            threadPoolExecutor.execute(tasksFactory.createExecutionWorker(
-                    service, scenarioHandler.getScenario(), proxyHandler.getProxy()));
+
+        threadPoolExecutor.execute(this::runParallelExecute);
+
+    }
+
+    private void runParallelExecute() {
+        while (!threadPoolExecutor.isShutdown()){
+            threadPoolExecutor.execute(() -> service.execute(
+                    scenarioHandler.getScenario(),
+                    proxyHandler.getProxy()
+            ));
         }
     }
```

Comment:
The code exhibits an unbounded task queue problem, leading to the accumulation of over 200 million queued tasks within a few seconds. New tasks are submitted to the ThreadPoolExecutor without rate limiting or control, resulting in a backlog of tasks that can potentially exhaust system resources and degrade performance. I suppose, there is no need to create another thread to runParallelExecute().
![Screenshot_20231031_130613](https://github.com/GFLCourses5/worker/assets/20342728/e01f9b42-f841-42ae-aa7b-a539597fa646)
![Screenshot_20231031_133046](https://github.com/GFLCourses5/worker/assets/20342728/c5a9ef24-6345-4a79-846d-f8e4f0852aec)
![Screenshot_20231031_134209](https://github.com/GFLCourses5/worker/assets/20342728/486e6f93-074b-434f-9f94-1b20364aa1f9)

Reply:
Yes but it more complex problem, than it's looks like, and if this pull will be accepted, I'm going to offer some ways of solutions to this problem.